src/components/Button/MucButton.vue

```diff
@@ -77,38 +92,20 @@ const emit = defineEmits<
   (e: "click") => void
 >();
 
-const buttonVariantClass = computed(() => {
-  switch (variant) {
-    case "secondary":
-      return "m-button--secondary";
-    case "ghost":
-      return "m-button--ghost";
-    default:
-      return "m-button--primary";
-  }
-});
-
-const iconAnimatedClass = computed(() => {
-  if (iconAnimated && iconShownLeft) {
-    return "m-button--animated-left";
-  } else if (iconAnimated) {
-    return "m-button--animated-right";
-  } else {
-    return "";
-  }
-});
+const animateIconSpin = ref(false);
 
 /**
  * Emit a click event if not in disabled state.
  */
 const handleClick = () => {
+  animateIconSpin.value = true;
+
   if (!disabled) emit("click");
-};
 
-/**
- * Ensure that the disabled button style is applied when it is in the 'disabled' state.
- */
-const disabledClass = computed(() => (disabled ? "disabled" : ""));
+  setTimeout(() => {
+    animateIconSpin.value = false;
+  }, 1000);
+};
```

Comment:
Is it necessary to do the timeout every time, or is it sufficient in the case of spinIconOnClick?

Reply:
no! fixed with [63dc965](https://github.com/it-at-m/muc-patternlab-vue/pull/582/commits/63dc96567b7c7355149d3bb65210e11b36fbf48e)